### PR TITLE
{tools} [GNU-4.9.3-2.25] lftp 4.6.4 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**
I know it is big, but mainly versionbumps. 
Needs ~~https://github.com/hpcugent/easybuild-framework/pull/1505~~

### CODE REVIEW

**Comment by boegel** [comment]
@hajgato: maybe this should wait until https://github.com/hpcugent/easybuild-easyconfigs/pull/2108 goes in, which basically replaces `GNU` by `GCC`, delivering both GCC & binutils

**Comment by hajgato** [comment]
@boegel Its fine for me, but oh boy, that will make a mess in existing modules.....

**Comment by boegel** [comment]
@hajgato: well, it depends I guess... You could choose to stick to using GNU, which is not going to disappear.

The intention is to merge the GCCcore stuff without affecting people with existing module stacks. That is, they can switch to using `GCC` as a toolchain to replace `GNU`, which is really `GCC` (via GCCcore) + `binutils` (rather than only `GCC`).

So, maybe that implies we shouldn't put this on hold at all. :-)

**Comment by hajgato** [comment]
@boegel: its up to you. But I will change (locally) from `GNU`->`GCC` only in `2016a`+ stacks...

**Comment by boegel** [comment]
@hajgato: yes, that makes a lot of sense, if the GCCcore makes it in, I plan to keep `GNU` out of the `2016a` toolchains

**Comment by boegel** [comment]
@wpoely86: please review/test?

File: easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+toolchain = {'name': 'GNU', 'version': '4.9.3-2.25'}
+
+source_urls = ['ftp://ftp.gnutls.org/gcrypt/gnutls/v%(version_major_minor)s']
+sources = [SOURCE_TAR_XZ]
+
+guilever = '1.8.8'
+guileshortver = '.'.join(guilever.split('.')[:2])
+dependencies = [
+    ('GMP', '6.0.0a'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb]: 6.1?

File: easybuild/easyconfigs/g/guile/guile-1.8.8-GNU-4.9.3-2.25.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+description = """Guile is the GNU Ubiquitous Intelligent Language for Extensions,
+ the official extension language for the GNU operating system."""
+
+toolchain = {'name': 'GNU', 'version': '4.9.3-2.25'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [GNU_SOURCE]
+
+dependencies = [
+    ('libtool', '2.4.6'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/guile/guile-1.8.8-GNU-4.9.3-2.25.eb]: dep or builddep?

File: easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb (hunk 1 above)
Comment (hajgato) [inline easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb]: @wpoely86 Lots of exeisting deps have `GMP 6.0.0a` deps. I think the benefits of bumping to `GMP 6.1` do not compensate the compications/effort. (bunch of new deps with `-GMP-6.1` suffix, dor example)

File: easybuild/easyconfigs/g/guile/guile-1.8.8-GNU-4.9.3-2.25.eb (hunk 2 above)
Comment (hajgato) [inline easybuild/easyconfigs/g/guile/guile-1.8.8-GNU-4.9.3-2.25.eb]: dep:
```
ldd /apps/brussel/ivybridge/software/guile/1.8.8-GNU-4.9.3-2.25/bin/guile 
        linux-vdso.so.1 =>  (0x00002aaaaaaab000)
        libguile.so.17 => /apps/brussel/ivybridge/software/guile/1.8.8-GNU-4.9.3-2.25/lib/libguile.so.17 (0x00002aaaaaaad000)
        libgmp.so.10 => /apps/brussel/ivybridge/software/GMP/6.0.0a-GNU-4.9.3-2.25/lib/libgmp.so.10 (0x00002aaaaabac000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x0000003dd8800000)
   ---->libltdl.so.7 => /apps/brussel/ivybridge/software/libtool/2.4.6-GNU-4.9.3-2.25/lib/libltdl.so.7 (0x00002aaaaac33000)
        libdl.so.2 => /lib64/libdl.so.2 (0x0000003dd6800000)
        libm.so.6 => /lib64/libm.so.6 (0x0000003dd5c00000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x0000003dd6400000)
        libc.so.6 => /lib64/libc.so.6 (0x0000003dd6000000)
        /lib64/ld-linux-x86-64.so.2 (0x0000003dd5800000)
        libfreebl3.so => /lib64/libfreebl3.so (0x0000003dd8c00000
```

File: easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/g/gnutls/gnutls-3.4.7-GNU-4.9.3-2.25.eb]: agreed, fine to keep as is if it causes dependency hell

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/5a70cdadef58f4bf3606 for a full test report.

**Comment by boegel** [comment]
@hajgato: doesn't work for me, see (partial) test report

Here's the problem:

```
configure: error:                                                                                                                                                              ***                                                                                                                                                                          *** Libnettle 3.1 was not found. 
```

**Comment by hajgato** [comment]
@boegel: For me it did when framework did not set `PKG_CONFIG_PATH` (when `pkgconfig` directory is in `lib64` directory).
 Was the framework up-to date? Does `nettle` module sets `PKG_CONFIG_PATH` like this:
```
prepend-path    PKG_CONFIG_PATH         $root/lib64/pkgconfig
```

**Comment by boegel** [comment]
@hajgato: ah, that may explain it, I'll retest

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e11d4c8e4eb245fcfa9e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a5b6b1bd0993058e2097 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**